trio/_channel.py

```diff
@@ -170,15 +198,16 @@ async def send(self, value):
         self._state.send_tasks[task] = value
         task.custom_sleep_data = self
 
-        def abort_fn(_):
+        def abort_fn(_) -> Abort:
```

Comment:
Should the argument be typed here?

Reply:
Yes, but I gave up on figuring out what it should be :sweat_smile:

I started trying to figure out `trio.lowlevel.wait_task_rescheduled` but I had a lot of trouble with it and ultimately decided to postpone tackling lowlevel - since at least in this case it doesn't matter when the parameter is unused.
[looking at trio-typing](https://github.com/python-trio/trio-typing/blob/d5835f7065f58ebdd95ca27b09d1d5cbe686b8b8/trio-stubs/lowlevel.pyi#L131-L133) i think it should be `Callable[[], NoReturn]`, though don't ask me why.

Also see discussion further down on the `type: ignore`

Reply:
We could always just do `_: object` :^)

(in all seriousness I'll take a look and see eventually:tm:)